File: ruby/binary/binary.rb
class Binary
  def initialize(b)
    raise ArgumentError if b =~ /[^01]/
    @digits = b.split('').map(&:to_i).reverse
  end

  attr_reader :digits
  def to_decimal
    dec = 0
    @digits.each.with_index do |d, i|
      dec += d * 2 ** i
    end
    dec
  end

end

module BookKeeping
  VERSION = 2
end

File: javascript/binary/binary.js
var Binary = function(code) {
  this.toDecimal = function() {
    if (code.match(/[^01]/)) return 0;
    var i = 0;
    return code.split('').reverse().reduce(function(sum, x) {
      i++;
      return sum + (Math.pow(2, (i - 1)) * x);
    }, 0)
  };
};

module.exports = Binary;
